bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/protocol/QbusCommunication.java

```diff
@@ -121,26 +121,29 @@ public synchronized void startCommunication() throws IOException, InterruptedExc
         InetAddress addr = InetAddress.getByName(handler.getAddress());
         Integer port = handler.getPort();
 
-        Socket socket = null;
+        if (port == null) {
+            handler.bridgeOffline(ThingStatusDetail.CONFIGURATION_ERROR, "Please set a correct port.");
+            return;
+        }
 
-        if (port != null) {
-            try {
-                socket = new Socket(addr, port);
-                qSocket = socket;
-                qOut = new PrintWriter(socket.getOutputStream(), true);
-                qIn = new BufferedReader(new InputStreamReader(socket.getInputStream()));
-            } catch (IOException e) {
+        if (addr == null) {
+            handler.bridgeOffline(ThingStatusDetail.CONFIGURATION_ERROR, "Please set the hostname of the Qbus server.");
+            return;
+        }
 
-            } finally {
-                if (socket == null) {
-                    Integer serverCheck = handler.getServerCheck();
-                    handler.bridgeOffline(ThingStatusDetail.COMMUNICATION_ERROR,
-                            "No communication with Qbus Server, will try to reconnect every " + serverCheck
-                                    + " minutes");
-                    return;
-                }
-            }
-        } else {
+        Socket socket = null;
+
+        try {
+            socket = new Socket(addr, port);
+            qSocket = socket;
+            qOut = new PrintWriter(socket.getOutputStream(), true);
+            qIn = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+        } catch (IOException e) {
+            String msg = e.getMessage();
+            logger.warn("Could not start listening thread, IOException: {}", msg);
+            Integer serverCheck = handler.getServerCheck();
+            handler.bridgeOffline(ThingStatusDetail.COMMUNICATION_ERROR,
+                    "No communication with Qbus Server, will try to reconnect every " + serverCheck + " minutes");
```

Comment:
Setting the bridge offline already produces a log message, so you could remove your log message. You could also include the exception's message in the Thing status detail message. The hint, that it will reconnect every x minutes is not necessary as this should be obligatory.

`git rebase` is only necessary if there are changes in the upstream repo, that you want to have in your PR to work with.